database/tx_queue.go

```diff
@@ -0,0 +1,49 @@
+package database
+
+import (
+	"time"
+	"sync"
+	
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+	"sort"
+)
+
+const awaitingTxsTimeInMs = 500
+
+type txQueue struct {
+	lastUpdate time.Time
+	txs        []*ethTypes.Transaction
+	mtx        sync.Mutex
+}
+
+func (q *txQueue) addTx(tx *ethTypes.Transaction) {
+	q.mtx.Lock()
+	defer q.mtx.Unlock()
+	q.lastUpdate = time.Now()
+	q.txs = append(q.txs, tx)
+	sort.Slice(q.txs, func(i, j int) bool { 
```

Comment:
I am thinking how to improve this, it's inefficient to sort it like this for every new TX especially for batch TXs. Maybe this could be extracted to a specific method that would get called between `add()` and `pop` only once. But it gets more complicated with new TXs comming in between sorting and popping out...